chain/chain/src/flat_storage_creator.rs

```diff
@@ -413,6 +413,12 @@ impl FlatStorageShardCreator {
             FlatStorageStatus::Disabled => {
                 panic!("initiated flat storage creation for shard {shard_id} while it is disabled");
             }
+            // If the flat storage is undergoing resharding it means it was previously created successfully,
+            // but resharding itself hasn't been finished and must be resumed
```

Comment:
Is the goal here to support interruption handling and resume for flat storage resharding? (e.g. if the node gets restarted midst flat storage resharding).

Reply:
Exactly